com.unity.perception/Runtime/Randomization/Randomizers/AssetSources/AssetSource.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using UnityEngine.Perception.Randomization.Samplers;
+
+namespace UnityEngine.Perception.Randomization
+{
+    /// <summary>
+    /// AssetSources are used to load assets from a generically within a <see cref="Randomizers.Randomizer"/>
+    /// </summary>
+    /// <typeparam name="T">The type of asset to load</typeparam>
+    [Serializable]
+    public sealed class AssetSource<T> where T : Object
```

Comment:
Pretty significant new feature here - there should also be tests and docs. Alternately, we could keep it on a branch and move it into a 'beta' period in which we put it through its paces.

Reply:
Tests and docs are certainly coming after the major PR changes have been approved ;)

Separate branch works too. I think that makes more sense considering that we'll probs still have to make a few changes based on our iteration with the services team as we integrate this into datamaker.